jsHelper/spicetifyWrapper.js

```diff
@@ -1248,481 +1269,376 @@ Spicetify.SVGIcons = {
-	constructor({ name, disabled = false, icon = undefined, trailingIcon = undefined, divider = false }) {
-		super();
-		this.name = name;
-		this.icon = icon;
-		this.trailingIcon = trailingIcon;
-		this.disabled = disabled;
-		this.divider = divider;
-		this.classList.add("main-contextMenu-menuItem");
-	}
-
-	render() {
-		const parseIcon = icon => {
-			if (icon && Spicetify.SVGIcons[icon]) {
-				return `<svg height="16" width="16" viewBox="0 0 16 16" fill="currentColor">${Spicetify.SVGIcons[icon]}</svg>`;
-			}
-			return icon || "";
-		};
-
-		this.innerHTML = `
-		<button class="main-contextMenu-menuItemButton ${this.disabled ? "main-contextMenu-disabled" : ""} ${
-			this.divider ? "main-contextMenu-dividerAfter" : ""
-		}">
-			${parseIcon(this.icon)}
-			<span class="ellipsis-one-line main-type-mesto main-contextMenu-menuItemLabel">${this.name}</span>
-			${parseIcon(this.trailingIcon)}
-		</button>`;
-	}
+Spicetify.ContextMenuV2 = (() => {
+	const registeredItems = new Map();
 
-	connectedCallback() {
-		if (!this.rendered) {
-			this.render();
-			this.rendered = true;
+	function parseProps(props) {
+		if (!props) return;
+		let uris = [];
+		let uids;
+		let contextUri;
+		if (props.uris) {
+			uris = props.uris;
+		} else if (props.uri) {
+			uris = [props.uri];
+		} else if (props.item?.uri) {
+			uris = [props.item.uri];
+		} else if (props.reference?.uri) {
+			uris = [props.reference.uri];
+		} else {
+			return;
+		}
+		if (props.uids) {
+			uids = props.uids;
+		} else if (props.uid) {
+			uids = [props.uid];
+		} else if (props.item?.uid) {
+			uids = [props.item.uid];
 		}
-	}
-
-	update(name, value) {
-		this[name] = value;
-		this.render();
-	}
-}
-
-class _HTMLContextSubmenu extends HTMLDivElement {
-	constructor({ items = [], placement = "bottom-start" } = {}) {
-		super();
-		this.items = items;
-		this.placement = placement;
-		this.style.zIndex = "9999";
-		this.style.position = "absolute";
-		this.style.inset = "0px auto auto 0px";
-	}
-	render() {
-		this._tippy = {
-			unmount: () => {},
-			popperInstance: {
-				forceUpdate: () => {}
-			}
-		};
 
-		const list = document.createElement("ul");
-		list.classList.add("main-contextMenu-menu");
-		list.append(...this.items);
-		this.append(list);
-
-		const { y: parentY, width: parentWidth } = this.parentElement.getBoundingClientRect();
-		const { width: thisWidth, height: thisHeight } = this.getBoundingClientRect();
-		let x = 0;
-		let y = this.parentElement.offsetTop;
-
-		switch (this.placement) {
-			case "top-start":
-			case "bottom-start":
-				x += parentWidth - 5;
-				break;
-			default:
-				x -= thisWidth - 5;
-				break;
-		}
-		const realY = y + parentY;
-		if (realY + thisHeight > window.innerHeight) {
-			y -= realY + thisHeight - window.innerHeight;
-		}
-		this.style.transform = `translate(${x}px, ${y}px)`;
-	}
+		if (props.contextUri) {
+			contextUri = props?.contextUri;
+		} else if (props.context?.uri) {
+			contextUri = props.context.uri;
+		}
```

Comment:
```suggestion
		const contextUri = props?.contextUri || props.context.uri || undefined;
```

Reply:
Wouldn't this change alter the behavior of this snippet? Shouldn't it be: `const contextUri = props.contextUri ?? props.context?.uri;` ? As:
1. props.context can be undefined
2. the original code doesn't check if props is undefined

Reply:
How so? 
Regarding 2 I didn't change that line: https://github.com/Delusoire/spicetify-cli/blob/master/jsHelper/spicetifyWrapper.js#L1276

Regarding 1, in the current code contextUri can also be undefined if you never define it if both conditionals fails.

Reply:
I realize that my wording was ambiguous. What I tried to say is that the new snippet isn't faithful to the original. The optional chaining of `props?.contextUri` is unnecessary because `props` is never undefined. And second, if `props.context` is undefined then accessing `props.context.uri` will produce an error, this null check was previously done inside the else if's condition. Also, the ` || undefined` is not necessary either.